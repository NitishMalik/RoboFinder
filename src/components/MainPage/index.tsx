import React, { Component } from "react";

import CardList from "../CardList";
import SearchBox from "../SearchBox";
import Scroll from "../Scroll";
import ErrorBoundary from "../ErrorBoundary";
import Header from "../Header";
import { IRobo } from "redux/interfaces";

interface IMainPageProps {
  getRobos: () => void;
  robots: IRobo[];
  searchField: string;
  onSearchChanged: (event: React.SyntheticEvent) => void;
  isPending: boolean;
}
export class MainPage extends Component<IMainPageProps, {}> {
  componentDidMount() {
    this.props.getRobos();
  }

  filterRobots = () => {
    const { robots, searchField } = this.props;
    return robots!.filter((robot) => {
      return robot.name!.toLowerCase().includes(searchField.toLowerCase());
    });
  };

  render() {
    const { onSearchChanged, isPending } = this.props;

    return (
      <div className="tc">
        <Header />
        <SearchBox onSearchInputChange={onSearchChanged} />
        <Scroll>
          {isPending ? (
            <h1>Loading</h1>
          ) : (
              <ErrorBoundary>
                <CardList robos={this.filterRobots()} />
              </ErrorBoundary>
            )}
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
