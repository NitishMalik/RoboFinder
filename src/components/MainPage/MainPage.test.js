import { shallow } from "enzyme";
import React from "react";
import MainPage from "./";

let wrapper;
beforeEach(() => {
  const mockProps = {
    searchField: "",
    robots: [],
    isPending: false,
    error: "",
    getRobos: jest.fn(),
  };

  wrapper = shallow(<MainPage {...mockProps} />);
});

test("render without crashing", () => {
  expect(wrapper.debug()).toMatchSnapshot();
});

test("filter Robots correctly", () => {
  const mockProps2 = {
    getRobos: jest.fn(),
    robots: [
      {
        id: 2,
        name: "John",
        email: "John@gmail.com",
      },
    ],
    searchField: "a",
    isPending: false,
  };
  const filtered = [];
  const wrapper2 = shallow(<MainPage {...mockProps2} />);
  expect(wrapper2.instance().filterRobots()).toEqual(filtered);
});
