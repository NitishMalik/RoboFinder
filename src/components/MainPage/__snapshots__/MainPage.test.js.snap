// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crashing 1`] = `
"<div className=\\"tc\\">
  <Header />
  <SearchBox searchChange={[undefined]} />
  <Scroll>
    <ErrorBoundary>
      <CardList robos={{...}} />
    </ErrorBoundary>
  </Scroll>
</div>"
`;
